app/packages/state/src/recoil/schemaSettings.atoms.ts

```diff
@@ -0,0 +1,311 @@
+
+export const schemaSearchTerm = atom<string>({
+  key: "schemaSearchTerm",
+  default: "",
+});
+
+export const showNestedFieldsState = atom<boolean>({
+  key: "showNestedFieldsState",
+  default: false,
+});
+
+export const schemaSelectedSettingsTab = atom<string>({
+  key: "schemaSelectedSettingsTab",
+  default: TAB_OPTIONS_MAP.SELECTION,
+});
+
+export const settingsModal = atom<{ open: boolean } | null>({
+  key: "settingsModal",
+  default: {
+    open: false,
+  },
+});
+
+export const allFieldsCheckedState = atom<boolean>({
+  key: "allFieldsCheckedState",
+  default: true,
+});
+
+export const expandedPathsState = atom<{} | null>({
+  key: "expandedPathsState",
+  default: null,
+});
+
+export const viewSchemaState = atom({
+  key: "viewSchemaState",
+  default: null,
+});
+
+export const fieldSchemaState = atom({
+  key: "fieldSchemaState",
+  default: null,
+});
+
+export const showMetadataState = atom({
+  key: "showMetadataState",
+  default: false,
+});
+
+export const includeNestedFieldsState = atom({
+  key: "includeNestedFieldsState",
+  default: true,
+});
+
+export const affectedPathCountState = atom({
+  key: "affectedPathCountState",
+  default: 0,
+});
+
+export const searchMetaFilterState = atom({
+  key: "searchMetaFilterState",
+  default: {},
+});
+
+export const lastAppliedPathsState = atom({
+  key: "lastAppliedExcludedPathsState",
+  default: {
+    excluded: [],
+    selected: [],
+  },
+});
+
+// tracks action and value - currently used for (un)select all action
+export const lastActionToggleSelectionState = atom<Record<
+  string,
+  boolean
+> | null>({
+  key: "lastActionToggleSelectionState",
+  default: null,
+});
+
+const getRawPath = (path: string) =>
+  path.startsWith("frames.") ? path.replace("frames.", "") : path;
+
+export const schemaSearchResultList = selector<string[]>({
+  key: "schemaSearchResultsSelector",
+  get: ({ get }) => get(schemaSearchResults),
+  set: ({ set, get }, newPaths) => {
+    if (newPaths instanceof DefaultValue) {
+      newPaths = [];
+    }
+    const viewSchema = get(viewSchemaState);
+    const fieldSchema = get(fieldSchemaState);
+    const combinedSchema = { ...fieldSchema, ...viewSchema };
+
+    const greenPaths = [...newPaths]
+      .filter((path) => {
+        const cleanPath = getRawPath(path);
+
+        return (
+          cleanPath &&
+          combinedSchema?.[cleanPath]?.ftype &&
+          !skipField(cleanPath, combinedSchema)
+        );
+      })
+      .map((path) => getRawPath(path));
+    set(schemaSearchResults, greenPaths);
+  },
+  cachePolicy_UNSTABLE: {
+    eviction: "most-recent",
+  },
+});
+
+export const schemaSearchResults = atom<string[]>({
+  key: "schemaSearchResults",
+  default: [],
+});
+
+export const selectedPathsState = atomFamily({
+  key: "selectedPathsState",
+  default: (param: {}) => param,
+  effects: [
+    ({ onSet, getPromise, setSelf }) => {
+      onSet(async (newPathsMap) => {
```

Comment:
I know this was discussed offline. Just noting again that this should probably be changed to a selector. The atom will have an invalid intermediary value of `newPathsMap`. `onSet` should only be used to update data external to recoil.